src/order/zorn.lean

```diff
@@ -252,6 +252,17 @@ let ⟨m, hm⟩ := @exists_maximal_of_nonempty_chains_bounded α (≤) _ h (λ a
 ⟨m, λ a ha, le_antisymm (hm a ha) ha⟩
 
 theorem zorn_partial_order₀ {α : Type u} [partial_order α] (s : set α)
+  (ih : ∀ c ⊆ s, chain (≤) c → ∃ ub ∈ s, ∀ z ∈ c, z ≤ ub) :
+  ∃ m ∈ s, ∀ z ∈ s, m ≤ z → z = m :=
```

Comment:
Isn't this statement defeq to `zorn_subset` below when `α = set _`? If so, can you replace the proof below with `zorn_partial_order₀ S`?

Reply:
You can, and similarly for the one below! Amended, thanks.